packages/cozy-realtime/src/CozyRealtime.js

```diff
@@ -0,0 +1,297 @@
+/* global WebSocket */
+import extend from 'lodash/extend'
+import pick from 'lodash/pick'
+import pickBy from 'lodash/pickBy'
+
+import { isSecureURL } from './helpers/url'
+import RealtimeSubscriptions from './RealtimeSubscriptions'
+import Validator, {
+  isBoolean,
+  isRequired,
+  isRequiredIfNo,
+  isString,
+  isURL
+} from './helpers/Validator'
+
+const MAX_RETRIES = 3
+const RETRY_BASE_DELAY = 1000
+
+export class CozyRealtime {
+  /**
+   * Log were subcribe messages sent are recored
+   * @type {Array}
+   */
+  _log = []
+
+  /**
+   * Promise of an opnened socket
+   * @type {Promise}
+   */
+  _socketPromise = null
+
+  /**
+   * Realtime subscriptions
+   * @type {RealtimeSubscriptions}
+   */
+  _subscriptions = null
+
+  _retries = 0
+  _retryDelay = RETRY_BASE_DELAY
+
+  /**
+   * Open a WebSocket
+   * @constructor
+   * @param {String}  domain        The cozy domain
+   * @param {Boolean} [secure=true] Indicates either the WebSocket should be
+   * secure or not
+   * @param {String}  token         The Application token
+   * @param {String}  url           URL of the cozy. Can be used in place of
+   * domain and secure parameters
+   */
+  constructor({ domain, secure = true, token, url, onError, onDisconnect }) {
+    Validator.create({
+      domain: [isRequiredIfNo(['url']), isString],
+      secure: [isBoolean],
+      token: [isRequired, isString],
+      url: [isRequiredIfNo(['domain']), isURL]
+    }).validate({ domain, secure, token, url })
+
+    this._domain = domain || new URL(url).host
+
+    this._secure = url ? isSecureURL(url) : secure
+    this._token = token
+    this._url = url
+
+    this._onError = onError
+    this._onDisconnect = onDisconnect
+
+    this._subscriptions = new RealtimeSubscriptions()
+    this._connect()
+  }
+
+  /**
+   * Returns an instance of CozyRealtime. Can be used instead of
+   * `new CozyRealtime`.
+   * @static
+   * @param  {Object} options Object containing domain, secure, token and url
+   * parameters
+   * @return {CozyRealtime}         CozyRealtime instance
+   */
+  static init(options) {
+    return new CozyRealtime(options)
+  }
```

Comment:
I think we should remove this. It is idiomatic javascript to write `new`, there is no advantage in having this method, and having several methods to do 1 thing is confusing IMHO.